src/main/java/teammates/logic/backdoor/BackDoorLogic.java

```diff
@@ -439,8 +439,13 @@ public void removeDataBundle(DataBundle dataBundle) {
         }
         accountsDb.deleteAccounts(dataBundle.accounts.values());
 
-        // Delete all admin emails
-        adminEmailsDb.deleteEntities(adminEmailsDb.getAllAdminEmails());
+        for (AdminEmailAttributes email : dataBundle.adminEmails.values()) {
+            AdminEmailAttributes emailInDb = adminEmailsDb.getAdminEmailBySubject(email.subject);
+            if (emailInDb == null) {
```

Comment:
2. Tests shouldnt be able to modify real data due to the changes in BackDoor. The adminEmails in the databundle is replaced with an empty map.